platform/commonUI/edit/src/creation/CreateAction.js

```diff
@@ -86,11 +86,23 @@ define(
                         })
                         .join('/');
 
-                openmct.router.navigate(url);
+                function editObject() {
+                    const path = objectPath.slice(-1).map(obj => {
+                        const objNew = obj.getCapability('adapter');
+                        objNew.identifier = openmct.objects.parseKeyString(obj.getId());
+
+                        return objNew;
+                    });
+                    if (isFirstViewEditable(object.useCapability('adapter'), path)) {
+                        openmct.editor.edit();
+                    }
 
-                if (isFirstViewEditable(object.useCapability('adapter'), objectPath)) {
-                    openmct.editor.edit();
+                    openmct.router.off('assigned:path', editObject);
                 }
+
+                openmct.router.on('assigned:path', editObject);
```

Comment:
You could use `openmct.router.once(` and void having to clean it up.

Reply:
Love it! Done.